addons/show-bbcode/userscript.js

```diff
@@ -0,0 +1,58 @@
+function viewSource(post, msg) {
+  return function (event) {
+    event.preventDefault();
+    const body = post.querySelector(".postmsg");
+    if (event.target.getAttribute("data-state") === "post") {
+      event.target.innerText = msg("source-button-active");
+      event.target.removeAttribute("title");
+      if (event.target.originalHTML === undefined) {
+        event.target.originalHTML = body.firstElementChild;
+      }
+      body.removeChild(body.firstElementChild);
+      const source = document.createElement("div");
+      body.insertBefore(source, body.firstElementChild);
+      source.className = "post_body_html";
+      if (event.target.sourceText !== undefined) {
+        event.target.setAttribute("data-state", "source");
+        source.innerText = event.target.sourceText;
+        return;
+      }
+      event.target.setAttribute("data-state", "loading");
+      source.innerText = msg("loading");
+      fetch("https://scratch.mit.edu/discuss/post/" + post.id.substring(1) + "/source/").then(function (res) {
```

Comment:
Wouldn't it be better to store this somewhere, so you don't have to have a slow request every single time?

Reply:
It's stored as `event.target.sourceText`. See lines 15 - 19.

Reply:
But there's nothing stopping you from clicking the button _while_ the state is loading.